src/dolphin/iniFile.ts

```diff
@@ -0,0 +1,277 @@
+/**
+ * rewrite of Inifile.cpp/Inifile.h from dolphin in Typescript
+ * https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Common/IniFile.cpp
+ */
+
+import log from "electron-log";
+import fs from "fs";
+import readline from "readline";
+
+/**
+ * The IniFile Class, contains a Section subclass
+ */
+export class IniFile {
+  private sections: Section[];
+
+  public constructor() {
+    this.sections = [];
+  }
+
+  /** Differs from IniFile.cpp via:
+   * Instead of editing keyOut and valueOut by reference, return them */
+  private parseLine(line: string): readonly [string, string] | readonly [null, null] {
+    let retValueOut = "";
+    let keyOut = "";
+
+    if (line === "" || line[0] === "#") {
+      return [null, null] as const;
+    }
+
+    const firstEquals = line.indexOf("=");
+    if (firstEquals !== -1) {
+      keyOut = line.substring(0, firstEquals).replace(/\s+/g, "");
+      retValueOut = line
+        .substring(firstEquals + 1)
+        .replace(/\s+/g, "")
+        .replace(/['"]+/g, "");
+    }
+
+    return [keyOut, retValueOut] as const;
+  }
+
+  /**Differs from IniFile.cpp by:
+   * returns section object, not pointer
+   */
+  public getSection(sectionName: string): Section | undefined {
+    const section = this.sections.find((section) => section.name === sectionName);
+    return section;
+  }
+
+  /**Differs from IniFile.cpp by:
+   * returns section object, not pointer
+   */
+  public getOrCreateSection(section_name: string): Section {
+    let section = this.getSection(section_name);
+    if (section === undefined) {
+      section = new Section(section_name);
+      this.sections.push(section);
+    }
+    return section;
+  }
+
+  public deleteSection(section_name: string): boolean {
+    const s = this.getSection(section_name);
+    if (s === undefined) {
+      return false;
+    }
+    this.sections.splice(this.sections.indexOf(s), 1);
+    return true;
+  }
+
+  public exists(section_name: string): boolean {
+    return this.getSection(section_name) != undefined;
+  }
+
+  public setLines(section_name: string, lines: string[]): void {
+    const section = this.getOrCreateSection(section_name);
+    section.setLines(lines);
+  }
+
+  public deleteKey(section_name: string, key: string): boolean {
+    const section = this.getSection(section_name);
+    if (section === undefined) {
+      return false;
+    }
+    return section.delete(key);
+  }
+
+  /**Differs from IniFile.cpp by:
+   * returns keys instead of passing it by reference
+   */
+  public getKeys(section_name: string): string[] {
```

Comment:
```suggestion
  public getKeys(sectionName: string): string[] {
```

There's still a lot of snake case so maybe it's best to just find and replace them all.